tools/scripts/update-bumpenvs-yaml.py

```diff
@@ -197,11 +202,16 @@ def update_tag_for_image_type(subconf: Dict[str, str], new_tag: str) -> bool:
         **yaml.safe_load(artifacts["publish-tf1-gpu"]),
         **yaml.safe_load(artifacts["publish-tf2-cpu"]),
         **yaml.safe_load(artifacts["publish-tf2-gpu"]),
+        **yaml.safe_load(artifacts["publish-cuda-11"]),
         **parse_packer_log(artifacts["packer-log"]),
     }
 
     saw_change = False
     for image_type, new_tag in new_tags.items():
+        if image_type not in conf:
+            # Add a placeholder to use as the new "old" image.
+            # This string should not appear elsewhere in the codebase.
+            conf[image_type] = {"new": NON_EXISTENT_IMAGE}
         saw_change |= update_tag_for_image_type(conf[image_type], new_tag)
```

Comment:
What about:
```python
if image_type not in conf:
    conf[image_type] = {"new": new_tag}
    saw_change = True
else:
    saw_change |= update_tag_for_image_type(conf[image_type], new_tag)
```

This is a case I never thought to test, but the case where `old` is empty is properly handled, and I'd rather just leave it empty when you introduce a new image.

Reply:
It's not properly handled everywhere - it definitely fell over when trying to deal with the new CUDA 11 images.

Reply:
Where else would it need to be handled?

Reply:
Oh I misunderstood the comment, nevermind. Yes I like your solution - quite elegant. Updated the PR.